plugin/pkg/proxy/connect.go

```diff
@@ -121,6 +122,11 @@ func (p *Proxy) Connect(ctx context.Context, state request.Request, opts Options
 			if err == io.EOF && cached {
 				return nil, ErrCachedClosed
 			}
+			// if the error is not the result of a timeout, continue waiting for an answer in case this was a spoofed
+			// malformed response.
+			if err, ok := err.(net.Error); ok && !err.Timeout() {
+				continue
+			}
```

Comment:
were you able to test this scenario somehow? I have doubts that this will work as the[ connection is closed few lines above](https://github.com/coredns/coredns/pull/6014/files#diff-58178efbf2d640db6d59880c39fc50089881ad7d4e117181b689b1eb4d9577d0R121)

Reply:
Thanks for taking a look. No, I’m still working on a test, but my bandwidth for coredns is limited recently and this keeps getting bumped off my plate.

Reply:
Coming back to this to move the close connection step ... I realize also that this proposed fix will cause catastrophic  problems for non-timeout connection errors.

I'm going to close this, since I'm not going to have time in the near future to give this the focus it requires.